src/subsystems/haskell/translators/stack-lock/utils.nix

```diff
@@ -16,16 +16,29 @@ in rec {
     tar --strip-components 1 -xf ${pkgs.all-cabal-hashes}
   '';
 
-  cabal2jsonSrc = builtins.fetchTarball {
-    url = "https://github.com/NorfairKing/cabal2json/tarball/8b864d93e3e99eb547a0d377da213a1fae644902";
-    sha256 = "0zd38mzfxz8jxdlcg3fy6gqq7bwpkfann9w0vd6n8aasyz8xfbpj";
-  };
-
-  cabal2jsonFlake = flakeCompat {
-    src = cabal2jsonSrc;
-  };
-
-  cabal2json = cabal2jsonFlake.defaultNix.packages.${pkgs.system}.cabal2json;
+  cabal2json = let
+    haskellPackages = pkgs.haskell.packages.ghc8107.override {
+      overrides = curr: prev: {
+        autodocodec = prev.autodocodec.overrideAttrs (old: {
+          meta = old.meta // {broken = false;};
+        });
+        validity-aeson = prev.validity-aeson.overrideAttrs (old: {
+          meta = old.meta // {broken = false;};
```

Comment:
There is a function called `markUnbroken` https://github.com/NixOS/nixpkgs/blob/93d3ee74a184a568b5a506ef76e6ffa00866440f/pkgs/development/haskell-modules/lib/compose.nix#L178

Never used it with flakes but it should work the same, right?

Reply:
Hi Michael, didn't know about these ones, nice!

Reply:
I see no reason why they shouldn't work (and they seems to work). Do you know if it's normal that one test failed?